pkg/applicationserver/io/packages/alcsync/v1/package.go

```diff
@@ -59,12 +58,15 @@ func (a *alcsyncpkg) HandleUp(
 
 	data, fPort, err := mergePackageData(def, assoc)
 	if err != nil {
-		logger.Error("Failed to merge package data")
+		logger.WithError(err).Error("Failed to merge package data")
```

Comment:
I wouldn't log this as `Error` to be honest - I think this should stay as debug.

The reason for this is that the package data is just a blob of JSON that comes from the user, and there is no validation on what they input here, so it is not a _server_ error if the contents are blaberred - what matters is that an event is generated.

For that matter, please take a look at how the `defer`ed error is used in other packages - you introduce a name return called `err` and inside the defer you generate the failed processing event.